From 12c5ab3b4aa8109fe3def3a67fe57206ebfd9b65 Mon Sep 17 00:00:00 2001
From: Shreenidhi Shedi <sshedi@vmware.com>
Date: Thu, 5 Nov 2020 01:34:18 -0800
Subject: [PATCH] Add users section of cloud config yaml only if password field
 is present

Don't clean cloud-init logs before invoking init

Signed-off-by: Shreenidhi Shedi <sshedi@vmware.com>
---
 gosc-scripts/CustomizationUtils.sh  | 7 +++++--
 gosc-scripts/PhotonCustomization.sh | 2 +-
 2 files changed, 6 insertions(+), 3 deletions(-)

diff --git a/gosc-scripts/CustomizationUtils.sh b/gosc-scripts/CustomizationUtils.sh
index e36547e..5dfefa6 100644
--- a/gosc-scripts/CustomizationUtils.sh
+++ b/gosc-scripts/CustomizationUtils.sh
@@ -961,17 +961,20 @@ hostname: ${hostname}
 EOF
 )
 
-  export formatResult=$formatResult$(${CAT} <<EOF
+  local adminPwd=$(ConfigFile_GetAdminPassword)
+
+  if [[ -n "$adminPwd" ]]; then
+    export formatResult=$formatResult$(${CAT} <<EOF
 
 users:
     - name: root
       lock_passwd: false
 EOF
 )
+  fi
 
   local ssh_key_cnt=$(ConfigFile_GetSshKeysCnt)
   if [ "$ssh_key_cnt" -gt 0 ]; then
-    local adminPwd=$(ConfigFile_GetAdminPassword)
     if [[ -z "$adminPwd" ]]; then
       export formatResult=$formatResult$(${CAT} <<EOF
 
diff --git a/gosc-scripts/PhotonCustomization.sh b/gosc-scripts/PhotonCustomization.sh
index 3a92712..1887db5 100644
--- a/gosc-scripts/PhotonCustomization.sh
+++ b/gosc-scripts/PhotonCustomization.sh
@@ -80,7 +80,7 @@ RunCloudConfig()
   fi
 
   sleep 2
-  cloud-init clean -ls
+  cloud-init clean -s
 
   GenerateCloudInitConfig content "GetInterfaceByMacAddressIPAddrShow" $tmpOvfEnvFile
   # Once the cloud-config file been generated, remove the existing network config files.
-- 
2.29.2

